main.cpp

```diff
@@ -12,13 +12,26 @@ int main(int argc, char* argv[])
     QCoreApplication app(argc, argv);
     auto args = app.arguments();
     bool use_gpu = true;
+    int gpu_bitcount = 0;
+
+    int first_file_idx = 1;
+    while(args[first_file_idx].startsWith("--")) first_file_idx++;
 
     if(args.contains("--cpu_only")) {
```

Comment:
I would include this arg check into the next loop.

Reply:
Good point, I should just get rid of that 1-line while loop above and merge the loops.